### PR TITLE
Fix authentication failure in case no credential store was added to SPS SSH connection

### PR DESCRIPTION
**Pull request by imgulyas**

### CODE REVIEW

File: src/common-ssh/ssh.c [hunk 1]
```diff
@@ -383,45 +395,23 @@ static int guac_common_ssh_authenticate(guac_common_ssh_session* common_session)
...
-                        GUAC_PROTOCOL_STATUS_CLIENT_UNAUTHORIZED,
-                        "Keyboard-interactive authentication failed: %s",
-                        error_message);
-
-                return 1;
+            int result = libssh2_userauth_keyboard_interactive(session, user->username,
+                        &guac_common_ssh_kbd_callback);
+
+            if (result == LIBSSH2_ERROR_AUTHENTICATION_FAILED) {
+                /* SPS changes to password authentication at this point */
```
Comment (krajorama) [inline src/common-ssh/ssh.c]: SPS -> server

File: src/common-ssh/ssh.c [hunk 2]
```diff
@@ -383,45 +395,23 @@ static int guac_common_ssh_authenticate(guac_common_ssh_session* common_session)
...
-                        GUAC_PROTOCOL_STATUS_CLIENT_UNAUTHORIZED,
-                        "Password authentication failed: %s", error_message);
-
-                return 1;
-            }
-
-            /* Password authentication succeeded */
-            return 0;
-
+            return password_auth_with_retry(common_session);
```
Comment (krajorama) [inline src/common-ssh/ssh.c:398]: We'll have to be backwards compatible eventually to push our patches upstream - ok for now. Meaning that we lose specifying the password as an input parameter at the moment.

File: src/common-ssh/ssh.c (hunk 1 above)
Comment (imgulyas) [reply src/common-ssh/ssh.c]: I guess I take it out from the commit message as well.

File: src/common-ssh/ssh.c (hunk 2 above)
Comment (imgulyas) [reply src/common-ssh/ssh.c:398]: ok

**Review by krajorama — APPROVED** [review]
lgtm, lets test

**Pull request merged**